python/ray/ml/train/integrations/huggingface/huggingface_trainer.py

```diff
@@ -146,52 +220,297 @@ def trainer_init_per_worker(train_dataset, eval_dataset, **config):
+            resume_from_checkpoint: Checkpoint = kwargs["resume_from_checkpoint"]
+            (
+                checkpoint_type,
+                checkpoint_path,
+            ) = resume_from_checkpoint.get_internal_representation()
+            if checkpoint_type != "local_path":
+                checkpoint_path = resume_from_checkpoint.to_directory()
+            if checkpoint_path:
+                # Load checkpoint from path.
+                checkpoint_path = Path(checkpoint_path).expanduser().absolute()
+                if not checkpoint_path.exists():
+                    raise ValueError(
+                        f"Checkpoint path {checkpoint_path} does not exist."
+                    )
+                with open(checkpoint_path.joinpath(TUNE_CHECKPOINT_ID), "r") as f:
+                    tune_checkpoint_id = int(f.read())
+
+                kwargs["resume_from_checkpoint"] = Checkpoint.from_dict(
+                    {
+                        NODE_IP_KEY: get_node_ip_address(),
+                        CHECKPOINT_PATH_ON_NODE_KEY: str(checkpoint_path),
+                        TUNE_CHECKPOINT_ID: tune_checkpoint_id,
+                    }
+                )
+        return super(HuggingFaceTrainer, cls).__new__(cls, *args, **kwargs)
+
+    def _validate_trainer_init_per_worker(
+        self, trainer_init_per_worker: Callable, fn_name: str
+    ) -> None:
+        num_params = len(inspect.signature(trainer_init_per_worker).parameters)
+        if num_params < 3:
+            raise ValueError(
+                f"{fn_name} should take in at least 3 arguments, "
+                f"but it accepts {num_params} arguments instead."
+            )
+
+    def _validate_train_loop_per_worker(
+        self, train_loop_per_worker: Callable, fn_name: str
+    ) -> None:
+        # Do not validate train_loop_per_worker. We validate
+        # trainer_init_per_worker instead.
+        pass
+
+    def _create_train_func(
+        self,
+        trainer_init_per_worker: Callable[
+            [TorchDataset, Optional[TorchDataset], Any], transformers.trainer.Trainer
+        ],
+    ):
         def train_loop_per_worker(config):
-            # Set to None just to make CI pass & show
-            # the intended usage with trainer_init_per_worker
-            train_dataset = None
-            eval_dataset = None
-            trainer = trainer_init_per_worker(train_dataset, eval_dataset, **config)
-            trainer.train()
+            # Env vars necessary for HF to setup DDP
+            os.environ["RANK"] = str(train.world_rank())
+            os.environ["WORLD_SIZE"] = str(train.world_size())
+            os.environ["LOCAL_RANK"] = str(train.local_rank())
+
+            train_dataset = train.get_dataset_shard(TRAIN_DATASET_KEY)
+            eval_dataset = train.get_dataset_shard(EVALUATION_DATASET_KEY)
+            train_columns = set(train_dataset.schema(fetch_if_missing=True).names)
+            if "labels" not in train_columns:
```

Comment:
From the documentation it seems like Huggingface is able to support any arbitrary Torch datasets. Can we leave the validation and error handling to Huggingface instead of doing it ourselves? This will make sure our implementation is future-proof.